src/errors.rs

```diff
@@ -268,222 +275,344 @@ impl Reportable for DocumentError {
+            .capabilities
+            .render
+            .as_ref()
+            .ok_or(HowserError::CapabilityError)?
+            .render_commonmark();
+        let rx_snippet = rx_node
+            .capabilities
+            .render
+            .as_ref()
+            .ok_or(HowserError::CapabilityError)?
+            .render_commonmark();
 
-        Ok(LinkError {
+        Ok(TextualContentError {
+            node_type,
             rx_file,
             rx_line,
-            document_file,
-            document_line,
+            doc_file,
+            doc_line,
+            rx_prompts,
+            doc_matches,
+            rx_snippet,
+            doc_snippet,
         })
     }
 }
 
-impl Reportable for LinkError {
+impl Reportable for TextualContentError {
     fn short_msg(&self) -> String {
-        let error_type =
-            ShellText::ErrorColor(Box::new(ShellText::Literal("Link Error".to_string())));
-        let rx_info = ShellText::Underlined(Box::new(ShellText::Literal(format!(
-            "{} line {}",
-            self.rx_file, self.rx_line
-        ))));
-        let document_info = ShellText::Underlined(Box::new(ShellText::Literal(format!(
-            "{} line {}",
-            self.document_file, self.document_line
-        ))));
-        format!(
-            "{} :: {}, {}",
-            error_type.to_string(),
-            rx_info.to_string(),
-            document_info.to_string()
-        )
+        let error_type = format!(
+            "{}Textual Content Error{}",
+            color::Fg(color::Red),
+            color::Fg(color::Reset)
+        );
+        let rx_info = format!(
+            "{}{} line {}{}",
+            style::Underline,
+            self.rx_file,
+            self.rx_line,
+            style::Reset
+        );
+        let doc_info = format!(
+            "{}{} line {}{}",
+            style::Underline,
+            self.doc_file,
+            self.doc_line,
+            style::Reset
+        );
+        let node_type = format!("{}{:?}{}", style::Bold, self.node_type, style::Reset);
+
+        format!("{}: {} at {}, {}", error_type, node_type, rx_info, doc_info)
     }
 
     fn long_msg(&self) -> String {
-        self.short_msg()
+        let mut message = format!(
+            "{}Textual Content Error{}\n\n",
+            color::Fg(color::Red),
+            color::Fg(color::Reset)
+        );
+        let rx_info = format!(
+            "{}{} line {}{}\n",
+            style::Underline,
+            self.rx_file,
+            self.rx_line,
+            style::Reset
+        );
+        let doc_info = format!(
+            "{}{} line {}{}\n",
+            style::Underline,
+            self.doc_file,
+            self.doc_line,
+            style::Reset
+        );
+
+        message += &format!("{}Prescription : {}", style::Bold, style::Reset);
+        message += &self.rx_prompts
+            .iter()
+            .flat_map(|s| s.chars())
+            .collect::<String>();
```

Comment:
This is concatenating the strings together, right? This cannot be fast; surely there's a better way.